pkgs/development/python-modules/cot/default.nix

```
@@ -85,5 +85,6 @@ buildPythonPackage rec {
     homepage = "https://github.com/glennmatthews/cot";
     license = licenses.mit;
     maintainers = with maintainers; [ evanjs ];
+    broken = !pythonOlder "3.12"; # Does not build
```

Can you please add a more verbose explanation of _ why_ it doesn't build?
 Also I've seen the last commit is five years ago. No packages (from a quick `rg`) inside `nixpkgs` use this tool. Maybe we should just remove it

---

1. The "more verbose explanation" I can think is just copy-paste the error log - a stack trace of functions that were probably removed in Python 3.12. I do not believe it would be so informative.
2. This package still works with a previous minor version of Python. It does not need to be removed right now.
3. Being a leaf package does not look a sufficient reason to completely drop a package.

On the other hand I believe this package has a minor CVE: https://github.com/glennmatthews/cot/pull/86

---

maybe just include `AttributeError: module 'configparser' has no attribute 'SafeConfigParser'`?
I guess we can think of removing it, when we move to python `3.13` :+1:

---

it is used here https://github.com/NixOS/nixpkgs/blob/b361c1b6a2edfe68a31aae39243a451a68f71716/nixos/modules/virtualisation/virtualbox-image.nix#L114

```suggestion
    broken = pythonAtLeast "3.12"; # not compatible because uses removed python packages and functions
```

---

Done

---

> I guess we can think of removing it, when we move to python `3.13` 👍

Only when Python 3.12 does not build anymore.

---

You mean `3,11` I guess 😉